crates/bevy_ecs/src/reflect.rs

```diff
@@ -265,7 +264,8 @@ pub struct ReflectResourceFns {
     /// Function pointer implementing [`ReflectResource::reflect()`].
     pub reflect: fn(&World) -> Option<&dyn Reflect>,
     /// Function pointer implementing [`ReflectResource::reflect_unchecked_mut()`].
-    pub reflect_unchecked_mut: unsafe fn(&World) -> Option<Mut<dyn Reflect>>,
+    /// SAFETY: The function may only be called with an InteriorMutableWorld that can be used to mutably access the relevant resource
```

Comment:
Nit: Formatting and linking
```suggestion
    ///
    /// # Safety
    /// The function may only be called with an [`InteriorMutableWorld`] that can be used to mutably access the relevant resource.
```